### PR TITLE
Ensure state cache evicts data after channel closes

### PR DESCRIPTION
**Pull request by uwiger**
- add mnesia table for persistent caching
- move cached data to mnesia once fsms terminate
- wait for 4 blocks after channel is gone from state tree
- fix multiple_channels test case bug (reusing same nonce)
- annotate fsms by test case for simpler debugging
- add gproc names for finding fsm by role or pubkey

### CODE REVIEW

**Comment by lucafavatella** [comment]
I see based on https://github.com/aeternity/epoch/pull/1364

File: apps/aechannel/src/aesc_fsm.erl [hunk 1]
```diff
@@ -1961,7 +1964,8 @@ start_min_depth_watcher(Type, SignedTx,
     evt({tx_hash, TxHash}),
     Nonce = aetx:nonce(Tx),
     evt({nonce, Nonce}),
-    {OnChainId, D1} = on_chain_id(D, Initiator, Nonce, Responder),
+    %% {OnChainId, D1} = on_chain_id(D, Initiator, Nonce, Responder),
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_fsm.erl]: Responder unused bound var?

Comment (lucafavatella) [inline apps/aechannel/src/aesc_fsm.erl]: This line can be deleted maybe?

File: apps/aechannel/src/aesc_state_cache.erl [hunk 2]
```diff
@@ -36,6 +57,25 @@ fetch(ChId, PubKey) ->
...
+    gen_server:call(?SERVER, {min_depth_achieved, ChId, self()}).
+
+table_specs(Mode) ->
+    [
+     {?MODULE, [
+                 aec_db:tab_copies(Mode)
+               , {type, ordered_set}
+               , {record_name, pch}
+               , {attributes, record_info(fields, pch)}
+               , {user_properties, [{vsn, table_vsn(pch)}]}
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_state_cache.erl]: I am a bit concerned that this `vsn` property is relied upon as infrastructure in `aec_db` so its definition shall not be here but in `aec_db`.

How does this definition of `vsn` here work with [`aec_db` code that checks `vsn` in existing schema](https://github.com/aeternity/epoch/blob/6bcfe0fa9fac56e928486b4095b94ac7b42d87cb/apps/aecore/src/aec_db.erl#L619-L627)? It looks like this table [gets ignored](https://github.com/aeternity/epoch/blob/6bcfe0fa9fac56e928486b4095b94ac7b42d87cb/apps/aecore/src/aec_db.erl#L602).

Maybe definition of this table shall be in `aec_db` as all other tables - less clean though simpler?

File: deployment/ansible/deploy.yml [hunk 3]
```diff
@@ -58,7 +58,7 @@
           port: 3114
       chain:
         persist: true
-        db_path: "./db67"
+        db_path: "./db68"
```
Comment (lucafavatella) [inline deployment/ansible/deploy.yml]: Would if be possible to add a line in release notes, for explaining DB incompatibility?

Comment (uwiger) [reply deployment/ansible/deploy.yml]: No one has started on the release notes for 0.19 yet, right?

File: apps/aechannel/src/aesc_state_cache.erl (hunk 2 above)
Comment (uwiger) [reply apps/aechannel/src/aesc_state_cache.erl]: There is a hook for checking tables. Adding the requisite code for it.

**Comment by uwiger** [comment]
Rebased and added dialyzer fixes and a check_tables hook.

**Comment by uwiger** [comment]
Rebased against master, changed merge base to master. Fixed test failures.

File: apps/aechannel/src/aesc_session_noise.erl [hunk 4]
```diff
@@ -95,7 +95,10 @@ accept(Port, Opts) ->
 init({Parent, Op}) ->
     %% trap exits to avoid ugly crash reports. We rely on the monitor to
     %% ensure that we close when the fsm dies
-    process_flag(trap_exit, true),
+    %%
+    %% TODO: For some reason, trapping exits causes
+    %% aehttp_integration_SUITE:sc_ws_open/1 to fail. Investigate why.
+    %% process_flag(trap_exit, true),
```
Comment (velzevur) [inline apps/aechannel/src/aesc_session_noise.erl]: Tracked as [Invesitgate aesc_session_noise process relations](https://www.pivotaltracker.com/story/show/159293763)

File: apps/aechannel/src/aesc_db.erl [hunk 5]
```diff
@@ -1,11 +1,18 @@
...
 
 create_tables(Mode) ->
     Specs = lists:flatten([M:table_specs(Mode) || M <- modules()]),
     [{atomic, ok} = mnesia:create_table(Tab, Spec)
      || {Tab, Spec} <- Specs].
 
+check_tables(Acc) ->
+    lists:foldl(
+      fun(M, Acc1) ->
+              M:check_tables(Acc1)
```
[the remark this answers is not in the document]
Comment (uwiger) [reply apps/aechannel/src/aesc_db.erl]: It is now. It was a premature push. :)

File: apps/aechannel/src/aesc_session_noise.erl [hunk 6]
```diff
@@ -87,6 +93,12 @@ accept(Port, Opts) ->
       ?MODULE, {self(), {accept, Port, Opts}}, ?GEN_SERVER_OPTS).
 
 init({Parent, Op}) ->
+    %% trap exits to avoid ugly crash reports. We rely on the monitor to
+    %% ensure that we close when the fsm dies
+    %%
+    %% TODO: For some reason, trapping exits causes
+    %% aehttp_integration_SUITE:sc_ws_open/1 to fail. Investigate why.
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aechannel/src/aesc_session_noise.erl]: It is: [Invesitgate aesc_session_noise process relations](https://www.pivotaltracker.com/story/show/159293763)
I wrote a comment for it but somehow it got lost

[the remark this answers is not in the document]
Comment (uwiger) [reply apps/aechannel/src/aesc_session_noise.erl]: Yes, https://www.pivotaltracker.com/story/show/159293763

**Comment by uwiger** [comment]
There was an apparently intermittent test failure, which didn't show up when I re-ran the job. The test didn't seem at all related to this PR, so merging now.

**Pull request merged**